packages/ra-core/src/routing/useCreateInternalLink.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from 'react';
+
+import { Identifier } from '../types';
+import { useBasename } from './useBasename';
+
+export const useCreateInternalLink = () => {
+    const basename = useBasename();
+    return useCallback(
+        ({ resource, id, type }: CreateInternalLinkParams): string => {
+            switch (type) {
+                case 'list':
+                    return `${basename}/${resource}`;
```

Comment:
in some old apps, users will use basepath instead of resource. The result will be a path like '//posts'. Easy fix: remove double slashes here to facilitate backwards compatibility.